Lib/http/cookiejar.py

```diff
@@ -1190,11 +1190,14 @@ def domain_return_ok(self, domain, request):
     def path_return_ok(self, path, request):
         _debug("- checking cookie path=%s", path)
         req_path = request_path(request)
-        if not req_path.startswith(path):
-            _debug("  %s does not path-match %s", req_path, path)
-            return False
-        return True
+        if req_path == path:
+            return True
+        elif (req_path.startswith(path) and
+              (path.endswith("/") or req_path[len(path)] == "/")):
```

Comment:
Note to self during review : I haven't identified a case where there will be `IndexError` for `req_path[len(path)]` since this comes after `req_path.startswith(path)` and hence the length shouldn't be less . A safer check would be to use slice equivalent `req_path[len(path):len(path)+1]` ?

Reply:
> . A safer check would be to use slice equivalent req_path[len(path):len(path)+1] ?

This might be a better idea.